agent/agent_endpoint.go

```diff
@@ -1041,8 +1041,21 @@ func (s *HTTPServer) AgentRegisterService(resp http.ResponseWriter, req *http.Re
 	}
 
 	// Add the service.
-	if err := s.agent.AddService(ns, chkTypes, true, token, ConfigSourceRemote); err != nil {
-		return nil, err
+	removeExistingChecks := false
```

Comment:
I suggest renaming that variable to `replaceExistingChecks` here and also below.